packages/augur-core/source/contracts/BaseSimpleDex.sol

```diff
@@ -0,0 +1,182 @@
+pragma solidity 0.5.15;
+
+import "ROOT/IAugur.sol";
+import "ROOT/ISimpleDex.sol";
+import "ROOT/CashSender.sol";
+import "ROOT/libraries/token/IERC20.sol";
+import "ROOT/libraries/token/VariableSupplyToken.sol";
+import "ROOT/libraries/Initializable.sol";
+import "ROOT/libraries/ReentrancyGuard.sol";
+import "ROOT/libraries/math/SafeMathUint256.sol";
+
+
+contract BaseSimpleDex is Initializable, ReentrancyGuard, VariableSupplyToken, CashSender {
+    using SafeMathUint256  for uint256;
+
+    string constant public name = "Simple Decentralized Exchange";
+    string constant public symbol = "SDE";
+
+    IAugur public augur;
+    address public token;
+    IERC20 public cash;
+
+    uint256 public tokenReserve;
+    uint256 public cashReserve;
+    uint256 public blockNumberLast;
+    uint256 public tokenPriceCumulativeLast;
+
+    event Mint(address indexed sender, uint256 tokenAmount, uint256 cashAmount);
+    event Burn(address indexed sender, uint256 tokenAmount, uint256 cashAmount, address indexed to);
+    event Swap(address indexed sender, address indexed tokenIn, uint256 amountIn, uint256 amountOut, address indexed to);
+    event Sync(uint256 tokenReserve, uint256 cashReserve);
+
+    function transferToken(address _to, uint256 _value) private;
+
+    function getTokenBalance() public returns (uint256);
+
+    function transferCash(address _to, uint _value) private {
+        cashTransfer(_to, _value);
+    }
+
+    function getCashBalance() public view returns (uint256) {
+        return cash.balanceOf(address(this));
```

Comment:
`cashBalanceOf`?  Why wasn't this caught by the new tests that check for calling `cash` directly?

Reply:
because it just got merged to master ;p